app/script/event/EventRepository.js

```diff
@@ -579,39 +579,37 @@ z.event.EventRepository = class EventRepository {
    * @returns {Promise} Resolves with the saved record or boolean true if the event was skipped
    */
   _handleEvent(event, source) {
-    return this._handleEventValidation(event, source).then(validatedEvent =>
-      this._processEvent(validatedEvent, source)
-    );
+    return this._handleEventValidation(event, source)
+      .then(validatedEvent => this.processEvent(validatedEvent, source))
+      .catch(error => {
+        const isIgnoredError = EventRepository.CONFIG.IGNORED_ERRORS.includes(error.type);
+        if (!isIgnoredError) {
+          throw error;
+        }
+
+        return true;
```

Comment:
Can we actually return the `event` there?
Else we need to change the consumer side to properly handle that `true` (I had the case in my current devs where I triggers some other crashes because the consumer always expect an event

Reply:
Good point. Tried to not change anything from the previous version in that regard but that does not make much sense.